runner/lib/tasks/testnet.js

```diff
@@ -227,6 +220,46 @@ export const makeTasks = ({
         configP2p.seeds = seeds.join(',');
         configP2p.addr_book_strict = false;
         delete config.log_level;
+
+        if (!useStateSync) {
+          console.log('Fetching genesis');
+          const genesis = await fetchAsJSON(`${testnetOrigin}/genesis.json`);
+
+          fs.writeFile(
+            joinPath(chainStateDir, 'config', 'genesis.json'),
+            JSON.stringify(genesis),
+          );
+        } else {
+          console.log('Fetching state-sync info');
+          /** @type {any} */
+          const currentBlockInfo = await fetchAsJSON(
+            `http://${rpcAddrs[0]}/block`,
+          );
+          const stateSyncInterval =
+            Number(process.env.AG_SETUP_COSMOS_STATE_SYNC_INTERVAL) || 2000;
+          const trustHeight = Math.max(
+            1,
+            Number(currentBlockInfo.result.block.header.height) -
+              stateSyncInterval,
```

Comment:
This is pretty standard state-sync, which requires a "root of trust". This is basically configuring state-sync saying we'll trust any block after the given height (and corresponding hash). The goal is to give something recent enough so that it doesn't need to verify too many blocks hashes with the light client, but old enough that there exists a state-sync snapshot taken after that block height.

Most state-sync instructions/scripts out there basically do this "latest block height - some offset" math without giving much explanation as to why. This is what I've gathered from understanding how state-sync works.

I'll add a comment adding an explanation.